Instrumenter instrumentation version and schema url (#5752)
open-telemetry/opentelemetry-java-instrumentation: docs/contributing/using-instrumenter-api.md

An `Instrumenter` can be built from several smaller components. The following subsections describe
all interfaces that can be used to customize an `Instrumenter`.

### Set the instrumentation version and OpenTelemetry schema URL

Always set the instrumentation library version on the `Instrumenter`. You can do this by calling
the `setInstrumentationVersion()` method on the `InstrumenterBuilder`. By setting the version, you

> let users identify which version of your instrumentation produced the telemetry.

Reviewer: do we want to promote the version-in-file as public best practice?

Author: I wasn't sure about that - I thought that it'd be the best to recommend the declarative `.setInstrumentationVersion()` way of setting it, while the properties file is sort of a hack that we use internally in all our instrumentations.
On the other hand, if we plan to make instrumentation-api stable it means that the properties file naming scheme (and the version property name) will also become stable - if we can't make backwards incompatible changes we might as well document it right away.
WDYT?

Reviewer: I think it's a useful pattern to promote for protection against uberjars

An `Instrumenter` can be built from several smaller components. The following subsections describe
all interfaces that can be used to customize an `Instrumenter`.

> ### Set the instrumentation version and OpenTelemetry schema URL

Author: CC @theletterf - I've added a new section to the Instrumenter API doc, please take a look at this when you have a moment 🙏

An `Instrumenter` can be built from several smaller components. The following subsections describe
all interfaces that can be used to customize an `Instrumenter`.

### Set the instrumentation version and OpenTelemetry schema URL

By setting the instrumentation library version, you let users identify which version of your
instrumentation produced the telemetry. Make sure you always provide the version to
the `Instrumenter`. You can do this in two ways:

* By calling the `setInstrumentationVersion()` method on the `InstrumenterBuilder`.
* By making sure that the JAR file with your instrumentation library contains a properties file in
  the `META-INF/io/opentelemetry/instrumentation/` directory. You must name the file
  `${instrumentationName}.properties`, where `${instrumentationName}` is the name of the
  instrumentation library passed to the `Instrumenter#builder()` method. The file must contain a
  single property, `version`. For example:

    ```properties
    # META-INF/io/opentelemetry/instrumentation/my-instrumentation.properties
    version = 1.2.3
    ```

>   The `Instrumenter` automatically detects the properties file and determine the instrumentation
>   version based on its name.

Reviewer: ```suggestion
  The `Instrumenter` automatically detects the properties file and determines the instrumentation
  version based on its name.
```

An `Instrumenter` can be built from several smaller components. The following subsections describe
all interfaces that can be used to customize an `Instrumenter`.

### Set the instrumentation version and OpenTelemetry schema URL

> Make sure you always provide the instrumentation library version to the `Instrumenter`. You can do

Author: @theletterf I've added a few more paragraphs, it'd be great if you could proofread them too 🙏

this in two ways:

* By calling the `setInstrumentationVersion()` method on the `InstrumenterBuilder`.

> * By making sure the JAR with your instrumentation library contains a properties file with the
>   version. The file must be located in the `META-INF/io/opentelemetry/instrumentation/` directory,
>   and it must be named `${instrumentationName}.properties`, where `${instrumentationName}` is the
>   name of the instrumentation library that is passed to the `Instrumenter#builder()` method. The
>   file must contain a single property `version`, e.g.:

Reviewer 2: ```suggestion
* By making sure that the JAR file with your instrumentation library contains a properties file in
   the `META-INF/io/opentelemetry/instrumentation/` directory. You must name the file 
   `${instrumentationName}.properties`, where `${instrumentationName}` is the
   name of the instrumentation library passed to the `Instrumenter#builder()` method. The
   file must contain a single property, `version`. For example:
```

with 
  the version.

    ```properties
    # META-INF/io/opentelemetry/instrumentation/my-instrumentation.properties
    version = 1.2.3
    ```

>   The `Instrumenter` will automatically detect these properties files and determine the
>   instrumentation version based on its name.

Reviewer 2: ```suggestion
  The `Instrumenter` automatically detects the properties file and determine the
  instrumentation version based on its name.
```

By setting the version, you let users identify which version of your instrumentation produced the

> telemetry.

Reviewer 2: Since this explains the importance of doing this, I'd move this paragraph right after the heading, before "Make sure..."
